Adds registration functionality for "Student" role

ArseniyVasilko opened the pull request:
Adds registration functionality for "Student" role 
Fixes login to only allow "Admin" users 
Passwords are now null for non-registered users, and get set once the user registers
Slight changes to debug_utils to account for new registration validators.

ArseniyVasilko committed:
Resolve conflict

Adds registration functionality for "Student" role
Fixes login to only allow "Admin" users
Passwords are now null for non-registered users, and get set once the user registers

ArseniyVasilko committed:
Adds registration functionality for "Student" role
Fixes login to only allow "Admin" users
Passwords are now null for non-registered users, and get set once the user registers

Logeshsharma commented on app/views.py:
```diff
@@ -12,6 +14,33 @@
+        return redirect(url_for('home'))
+    form = RegisterForm()
+    if form.validate_on_submit():
+        user_to_check = db.session.scalar(db.select(User).where(User.student_id==form.student_id.data))
```
form.student_id.data - need to be type convert to int before comparison

ArseniyVasilko replied:
Not necessary - form field is of type "IntegerField()" and only validates/stores data as an integer already.

Logeshsharma replied:
Ok

Logeshsharma commented on app/views.py:
```diff
@@ -12,6 +14,33 @@
+        if user_to_check == None:
+            flash("Entered credentials do no match our records - please check and try again", "danger")
+            return redirect(url_for('registration'))
+        if user_to_check.username != form.username.data or user_to_check.email != form.email.data:
```
Here checking with existing email is sufficient i think, is that necessary to check the username as well?

ArseniyVasilko replied:
You're right it's not necessary - I couldn't choose whether to check against email or username and used both for additional verification/security. Should I remove username then?

Logeshsharma replied:
yes, we can remove the username condition, thanks

Logeshsharma commented on app/views.py:
```diff
@@ -12,6 +14,33 @@
+            return redirect(url_for('registration'))
+        user_to_check.password_hash = generate_password_hash(form.password.data)
+        user_to_check.registered = True
+        db.session.add(user_to_check)
```
We don't want to add the user again, we get the user from db, so we update and commit, no need to add the user to session

ArseniyVasilko replied:
You're right - I forgot that user_to_check is not an new object, so it;s not necessary to add it to the session before the commit. This is pointless line, but also completely harmless lol. Need to change?

Logeshsharma replied:
yeah, it won't harm, but lets follow the best practice. ;)

ArseniyVasilko committed:
Adds task creation functionality for the "Admin" role.
Once a task is created, it is automatically assigned to each existing group and tracked individually.

Logeshsharma committed:
Fix get group mobile response

Logeshsharma committed:
Merge pull request #6 from Logeshsharma/fix/get-group-response

Fix get group mobile response

ArseniyVasilko commented:
Accidentally copied previous commit message - actually meant:

Fixed minor errors from Logesh's review.
Added Student Registration link to the navbar

ArseniyVasilko committed:
Additional start_datetime, end_datetime and location fields added to the db model.
Same fields and respective validation added to the task form as well.
"Create A Task" link added to navbar (Admins only)

ArseniyVasilko committed:
Additional start_datetime, end_datetime and location fields added to the db model.
Same fields and respective validation added to the task form as well.
"Create A Task" link added to navbar (Admins only)

ArseniyVasilko committed:
Additional start_datetime, end_datetime and location fields added to the db model.
Same fields and respective validation added to the task form as well.
"Create A Task" link added to navbar (Admins only)

Logeshsharma committed:
Merge pull request #5 from Logeshsharma/feature/create_task

Adds task creation functionality for the "Admin" role.

Logeshsharma committed:
Merge remote-tracking branch 'origin/feature/registration' into feature/registration

# Conflicts:
#	app/data/data.sqlite
#	app/forms.py
#	app/views.py
